Sources/TuistSupport/Extensions/AbsolutePath+Extras.swift

```diff
@@ -59,45 +58,7 @@ extension AbsolutePath {
 
     /// Returns the hash of the file the path points to.
     public func sha256() -> Data? {
-        do {
-            let bufferSize = 1024 * 1024
-            // Open file for reading:
-            let file = try FileHandle(forReadingFrom: url)
-            defer {
-                file.closeFile()
-            }
-
-            // Create and initialize SHA256 context:
-            var context = CC_SHA256_CTX()
-            CC_SHA256_Init(&context)
-
-            // Read up to `bufferSize` bytes, until EOF is reached, and update SHA256 context:
-            while autoreleasepool(invoking: {
-                // Read up to `bufferSize` bytes
-                let data = file.readData(ofLength: bufferSize)
-                if data.count > 0 {
-                    data.withUnsafeBytes {
-                        _ = CC_SHA256_Update(&context, $0, numericCast(data.count))
-                    }
-                    // Continue
-                    return true
-                } else {
-                    // End of file
-                    return false
-                }
-            }) {}
-
-            // Compute the SHA256 digest:
-            var digest = Data(count: Int(CC_SHA256_DIGEST_LENGTH))
-            digest.withUnsafeMutableBytes {
-                _ = CC_SHA256_Final($0, &context)
-            }
-
-            return digest
-        } catch {
-            print(error)
-            return nil
-        }
+        return try? SHA256Digest.file(at: url)
```

Comment:
Just spotted there's already a `SHA256` utility in the `Basic` library which could serve as an alternate

Reply:
I added https://inneka.com/programming/swift/sha256-in-swift/ to the SHA256Diegest, as I see, implementation that I add is for files, there is a buffer of 4k bytes. Basic has an implementation for `Data`, so you read the all data and then generate sha. So, I think using input stream is more efficient.